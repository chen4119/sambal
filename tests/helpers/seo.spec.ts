import {
    serializeJsonLd,
    renderSocialMediaMetaTags
} from "../../src/helpers/seo";


describe("seo", () => {
    const baseUrl = "https://example.com";
    const article = {
        "@type": "Article",
        name: "name field",
        headline: "Title of my article",
        description: "Description about my article",
        mainEntityOfPage: "/article",
        text: "<have some characters to <escape>>",
        image: {
            "@type": "ImageObject",
            contentSize: "1 kB",
            contentUrl: "/images/image2-50.webp",
            encodingFormat: "image/webp",
            height: 50,
            width: 75
        }
    };

    describe("serializeJsonLd", () => {
        it('article', () => {
            const result = serializeJsonLd(article);
            expect(result).toMatchSnapshot();
        });
    });

    describe("renderSocialMediaMetaTags", () => {
        it('article', async () => {
            const result = await renderSocialMediaMetaTags(baseUrl, article);
            expect(result).toMatchSnapshot();
        });
    });

});