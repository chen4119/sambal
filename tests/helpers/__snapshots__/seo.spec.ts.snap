// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`seo renderSocialMediaMetaTags article 1`] = `
"
            
        <link rel=\\"canonical\\" href=\\"https://example.com/article\\" />
        <title>Title of my article</title>
    
            
        <meta name=\\"twitter:card\\" content=\\"summary\\" />
        <meta name=\\"twitter:title\\" content=\\"Title of my article\\" />
        <meta name=\\"twitter:description\\" content=\\"Description about my article\\" />
    
            
        <meta name=\\"og:url\\" content=\\"https://example.com/article\\" />
        <meta name=\\"og:title\\" content=\\"Title of my article\\" />
        <meta name=\\"og:description\\" content=\\"Description about my article\\" />
    
        "
`;

exports[`seo serializeJsonLd article 1`] = `
"{
    \\"@type\\": \\"Article\\",
    \\"name\\": \\"name field\\",
    \\"headline\\": \\"Title of my article\\",
    \\"description\\": \\"Description about my article\\",
    \\"image\\": {
        \\"@type\\": \\"ImageObject\\",
        \\"contentSize\\": \\"1 kB\\",
        \\"contentUrl\\": \\"/images/image2-50.webp\\",
        \\"encodingFormat\\": \\"image/webp\\",
        \\"height\\": 50,
        \\"width\\": 75
    }
}"
`;
